src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/HotReload/ProjectHotReloadSession.cs

```diff
@@ -73,13 +73,20 @@ public async Task<bool> ApplyLaunchVariablesAsync(IDictionary<string, string> en
             }
 
             return false;
+}
+
+        // TODO: remove when Web Tools is no longer calling this method.
+        public Task StartSessionAsync(CancellationToken cancellationToken)
+        {
+            return StartSessionAsync(runningUnderDebugger: false, cancellationToken);
         }
 
-        public async Task StartSessionAsync(CancellationToken cancellationToken)
+        public async Task StartSessionAsync(bool runningUnderDebugger, CancellationToken cancellationToken)
         {
             if (!_sessionActive)
             {
-                await _hotReloadAgentManagerClient.Value.AgentStartedAsync(this, cancellationToken);
+                HotReloadAgentFlags flags = runningUnderDebugger ? HotReloadAgentFlags.IsDebuggedProcess : HotReloadAgentFlags.None;
```

Comment:
Would it be better to have the caller pass these flags directly, rather than a boolean? Future flags wouldn't require an API change in that case.

Reply:
Part of the purpose of this type is to abstract away the details of the Hot Reload machinery; passing the flags themselves would defeat that.